### PR TITLE
[Serialization] Recover from a missing conforming type in a protocol conformance

### PR DESCRIPTION
**Pull request by xymus**
We saw this failure with a Clang module imported @_implementationOnly with synthesized conformances by the ClangImporter. It caused issues only in sil-opt as it reads all the witness tables.

rdar://problem/58924131

### CODE REVIEW

**Comment by xymus** [comment]
@swift-ci please smoke test

File: lib/Serialization/DeserializeSIL.cpp [hunk 1]
```diff
@@ -3108,8 +3118,12 @@ SILWitnessTable *SILDeserializer::readWitnessTable(DeclID WId,
   }
 
   // Deserialize Conformance.
+  auto maybeConformance = MF->readConformanceChecked(SILCursor);
+  if (!maybeConformance)
```
Comment (nkcsgexi) [inline lib/Serialization/DeserializeSIL.cpp]: could you add some comments why this error may happen? Otherwise LGTM.

**Review by nkcsgexi — APPROVED** [review]

**Review by slavapestov — COMMENTED** [review]
This needs a test case.

File: lib/Serialization/DeserializeSIL.cpp (hunk 1 above)
Comment (xymus) [reply lib/Serialization/DeserializeSIL.cpp]: I've added documentation on the prototype for readWitnessTableChecked and on the only client readWitnessTables.

**Comment by xymus** [comment]
@swift-ci Please smoke test

**Comment by xymus** [comment]
@swift-ci Please smoke test

**Comment by futurejones** [comment]
Can one of the admins verify this patch?

**Comment by xymus** [comment]
@swift-ci please smoke test

**Comment by xymus** [comment]
@swift-ci please test Windows platform

**Pull request merged**